packages/renderer/src/modules/server.ts

```diff
@@ -0,0 +1,85 @@
+import { MessageTransport, RPC } from '@dcl/mini-rpc';
+
+import { fs } from '#preload';
+import { type Project } from '/shared/types/projects';
+
+export enum Method {
+  READ_FILE = 'read_file',
+  WRITE_FILE = 'write_file',
+  EXISTS = 'exists',
+  DELETE = 'delete',
+  LIST = 'list',
+}
+
+export type Params = {
+  [Method.READ_FILE]: {
+    path: string;
+  };
+  [Method.WRITE_FILE]: {
+    path: string;
+    content: Buffer;
+  };
+  [Method.DELETE]: {
+    path: string;
+  };
+  [Method.EXISTS]: {
+    path: string;
+  };
+  [Method.LIST]: {
+    path: string;
+  };
+};
+
+export type Result = {
+  [Method.READ_FILE]: Buffer;
+  [Method.WRITE_FILE]: void;
+  [Method.DELETE]: void;
+  [Method.EXISTS]: boolean;
+  [Method.LIST]: {
+    name: string;
+    isDirectory: boolean;
+  }[];
+};
+
+export type Callbacks = { [key in Method]: (params: Params[key]) => Result[key] };
+
+export function initTransport(
+  target: HTMLIFrameElement,
+  project: Project,
+  cbs: Partial<Callbacks> = {},
+) {
+  const transport = new MessageTransport(window, target.contentWindow!);
+  const storage = new RPC<Method, Params, Result>('IframeStorage', transport);
+  storage.handle('read_file', async ({ path }) => {
+    const file = await fs.readFile(await fs.resolve(project.path, path));
+    return file;
+  });
+  storage.handle('write_file', async ({ path, content }) => {
+    await fs.writeFile(await fs.resolve(project.path, path), content);
+    cbs.write_file?.({ path, content });
```

Comment:
I know that this is to reuse the `Method` enum, but it's not correctly typed and also it forces us to use the `snake_case` notation which is inconsistent. I think I would rather write a few more lines and have a well typed interface and consistent casing, ie:

```tsx
interface Callback {
  writeFile?: (params: Params[Method.WRITE_FILE]) => Promise<Result[Method.WRITE_FILE]>
}
```